Corrieresults

AlexanderLyNL opened the pull request:
- Remove correlationBayesianPairs.qml
- Still need to fix up the Robustness plot dots colours
- I still need to add bcorPackage as a package on my own repository instead of as a separate file
- The latter will help us separate the name space of the old functions and the new packag functions

Wishlist:
- General matrix plot for descriptives, correlation, correlationbayesian
- When the matrix plot is big, the waity circle is hard to see @tim
- Match up functions between correlation and correlationbayesian
- Match up functions between correlationbayesian and ttests
- I need to make a wrapper for t-test and do some computations there to simplify stuff
- Perhaps use data.table for ggplot2 stuff. I suspect some speed issues there
- A separate plotRobustness functionality in JASPgraphs. My poor attempt is plotRobustnessCor, which is covfefe sad
- Make qml mapper or item dictionary to map the arguments of correlationbayesianOld and correlationbayesianpairs to correlationbayesian

AlexanderLyNL committed:
bCor main table Pearson done

- Footnotes need to be fixed by Joris
- Still need to remove Bayesian correlation pairs

AlexanderLyNL committed:
bCor main table Pearson done

- Footnotes need to be fixed by Joris
- Still need to remove Bayesian correlation pairs

AlexanderLyNL committed:
bCor main table Pearson done better

- .computeBCorOneSided could not find fit in scope
- Make changing ciValue available
- pass jaspResults in markup
- and fill also if not all variables are there, because compute function already safeguards.

AlexanderLyNL committed:
bCor main table Pearson done better

- .computeBCorOneSided could not find fit in scope
- Make changing ciValue available
- pass jaspResults in markup
- and fill also if not all variables are there, because compute function already safeguards.

AlexanderLyNL committed:
Failure: Table disappears when subdependency is changed

Table and matrixPlot in container

AlexanderLyNL committed:
Failure: Table disappears when subdependency is changed

Table and matrixPlot in container

AlexanderLyNL committed:
Stable pairs plots as well now

- Still look into computations and memory

AlexanderLyNL committed:
Stable pairs plots as well now

- Still look into computations and memory

AlexanderLyNL committed:
sumStats fix correlation + added example in jaspGraphs

TODO:
1. A specific robustness plot function in jaspGraphs
2. Fix the points in the robustness analysis
3. Info consistency case

This is right before Kendall upload

AlexanderLyNL committed:
sumStats fix correlation + added example in jaspGraphs

TODO:
1. A specific robustness plot function in jaspGraphs
2. Fix the points in the robustness analysis
3. Info consistency case

This is right before Kendall upload

AlexanderLyNL committed:
correlation done mod some stuff

TODO:
1. Predictive matching for kappa stuff
2. Points in robustness
3. Remove correlationbayesianpairs from ribbon
4. Solve .outputSink (I suspect that it's because of corBayesTable$showSpecifiedColumnsOnly <- TRUE

AlexanderLyNL committed:
correlation done mod some stuff

TODO:
1. Predictive matching for kappa stuff
2. Points in robustness
3. Remove correlationbayesianpairs from ribbon
4. Solve .outputSink (I suspect that it's because of corBayesTable$showSpecifiedColumnsOnly <- TRUE

AlexanderLyNL committed:
Corrie done, but still not happy with the robustness plot colours

TODO:
- Ask Don for help

AlexanderLyNL committed:
Corrie done, but still not happy with the robustness plot colours

TODO:
- Ask Don for help

AlexanderLyNL committed:
Prepared for PR

- Remove correlationBayesianPairs.qml
- Still need to fix up the Robustness plot dots colours
- I still need to add bcorPackage as a package on my own repository instead of as a separate file
- The latter will help us separate the name space of the old functions and the new packag functions

Wishlist:
- General matrix plot for descriptives, correlation, correlationbayesian
- When the matrix plot is big, the waity circle is hard to see @Tim
- Match up functions between correlation and correlationbayesian
- Match up functions between correlationbayesian and ttests
- I need to make a wrapper for t-test and do some computations there to simplify stuff
- Perhaps use data.table for ggplot2 stuff. I suspect some speed issues there
- A separate plotRobustness functionality in JASPgraphs. My poor attempt is plotRobustnessCor, which is covfefe sad
- Make qml mapper or item dictionary to map the arguments of correlationbayesianOld and correlationbayesianpairs to correlationbayesian

AlexanderLyNL committed:
Prepared for PR

- Remove correlationBayesianPairs.qml
- Still need to fix up the Robustness plot dots colours
- I still need to add bcorPackage as a package on my own repository instead of as a separate file
- The latter will help us separate the name space of the old functions and the new packag functions

Wishlist:
- General matrix plot for descriptives, correlation, correlationbayesian
- When the matrix plot is big, the waity circle is hard to see @Tim
- Match up functions between correlation and correlationbayesian
- Match up functions between correlationbayesian and ttests
- I need to make a wrapper for t-test and do some computations there to simplify stuff
- Perhaps use data.table for ggplot2 stuff. I suspect some speed issues there
- A separate plotRobustness functionality in JASPgraphs. My poor attempt is plotRobustnessCor, which is covfefe sad
- Make qml mapper or item dictionary to map the arguments of correlationbayesianOld and correlationbayesianpairs to correlationbayesian

AlexanderLyNL committed:
Fixes up bayesian t.test plots

Fixes the case where NULL is being used as an alias for "equal"

AlexanderLyNL committed:
Fixes up bayesian t.test plots

Fixes the case where NULL is being used as an alias for "equal"

AlexanderLyNL committed:
Restored JASPgraphs' plotPriorAndPosterior

Not mess around with:

Check whether Don's fix also fixes the label switching problem on the right top at the sequential analysis plot

AlexanderLyNL committed:
Restored JASPgraphs' plotPriorAndPosterior

Not mess around with:

Check whether Don's fix also fixes the label switching problem on the right top at the sequential analysis plot

koenderks committed:
Remove "Forgy" label for algorithm and change to "Lloyd-Forgy"

+ as requested in https://github.com/jasp-stats/INTERNAL-jasp/issues/570

AlexanderLyNL committed:
Fixed up unit tests, not actually really sure about the plots, but okay

AlexanderLyNL committed:
Fixed up unit tests, not actually really sure about the plots, but okay

TimKDJ commented on Resources/Summary Statistics/qml/SummaryStatsCorrelationBayesianPairs.qml:
```diff
@@ -24,58 +24,71 @@ import JASP.Widgets 1.0
-	usesJaspResults: false
+    usesJaspResults: true
 
-	IntegerField { name: "sampleSize"; label: qsTr("Sample size") }
```
Can you keep the names as is in the summary stats module? That way we don't need backwards compatibility stuff everywhere

TimKDJ commented on Resources/Regression/qml/CorrelationBayesianOld.qml:
```diff
@@ -0,0 +1,95 @@
+//
```
You can remove this entire file. Github is very good at preserving changes ;)

TimKDJ commented on Resources/Regression/qml/CorrelationBayesian.qml:
```diff
@@ -23,73 +23,118 @@ import JASP.Widgets 1.0
-		CheckBox { name: "spearman";		label: qsTr("Spearman"); debug: true		}
-		CheckBox { name: "kendallsTauB";	label: qsTr("Kendall's tau-b")				}
-	}
+    VariablesForm
```
Please use tabs for .qml files

TimKDJ commented on JASP-Engine/JASP/R/bCorPackage.R:
```diff
@@ -0,0 +1,2236 @@
+# 1. Main test function --------
+bcor.test <- function(x, y, alternative=c("two.sided", "less", "greater"),
```
I think you showed that this already a functional R package? If you host this on your github as a package then we can just do `install_github()` and put it in required packages. That way you keep complete control over your own package (i.e., you don't need to make pull requests to work on the package)

(I won't look at the package files further, your package, your code style ;))

TimKDJ commented on JASP-Engine/JASP/R/bfCommon.R:
```diff
@@ -0,0 +1,131 @@
+  return(itemNames)
+}
+
+.bCorRowNames <- function(options, itemNames, method=c("pearson", "kendall", "spearman")) {
```
Quite a few of these are very specific to correlation. Maybe move to your commoncorrelation file?

TimKDJ commented on JASP-Engine/JASP/R/bfCommon.R:
```diff
@@ -0,0 +1,131 @@
+}
+
+
+.bSelectItems <- function(options) {
```
I would have no idea what this does. A little more descriptive name/some documentation would help

TimKDJ commented on JASP-Engine/JASP/R/commoncor.R:
```diff
@@ -0,0 +1,320 @@
+  nPairs <- length(pairs)
+  
+  if (nPairs!=0) {
+    lastPair <- pairs[[nPairs]]
```
Any pair can be incomplete

TimKDJ commented on JASP-Engine/JASP/R/commoncor.R:
```diff
@@ -0,0 +1,320 @@
+  .plotScatter(xVar = x, yVar = y, xBreaks = xBreaks, yBreaks = yBreaks, xName = xName, yName = yName) 
+}
+
+# # TODO(Alexander): Remove this, since it's stolen from Simon
```
If this is not going to be used, remove (github will keep its copy of old code)

TimKDJ commented on JASP-Engine/JASP/R/correlationbayesianOld.R:
```diff
@@ -0,0 +1,2864 @@
+#
```
Remove the entire file

TimKDJ commented on JASP-Engine/JASP/R/summarystatscorrelationbayesianpairs.R:
```diff
@@ -8,1287 +8,244 @@
 # This program is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# MERCHANTABILITY or FITNESS FOR A P          ARTICULAR PURPOSE.  See the
```
Don't think this was intended

AlexanderLyNL commented:
See the points in the last commit, and in particular:

<img width="1360" alt="Screen Shot 2019-11-18 at 10 48 15" src="https://user-images.githubusercontent.com/10271675/69144717-32cb1e80-0acc-11ea-901c-d4ce89d8149a.png">

TimKDJ commented on JASP-Engine/JASP/R/correlationbayesian.R:
```diff
@@ -15,2850 +15,1312 @@
-  # For the likelihood we use Jeffreys's approximation 
-  #   rho \propto (1-rho^2)^((n-1)/2)*(1-rho*r)^((3-2*n)/2)
-  # 
-  # The log prior is (alpha-1)*log(1-rho^2) # where 
```
Noticed quite a few of these (`&`) -- considering you don't want to do elementwise comparisons, use `&&`.
Same for `|` in your if-statements.

TimKDJ commented on JASP-Engine/JASP/R/correlationbayesian.R:
```diff
@@ -15,2850 +15,1312 @@
-	return(result)
+  # 3. Table: Get, compute and fill -----------
+  #
+  allBfObjects <- .makeTableCorBayes("jaspResults"=jaspResults, "allBfObjects"=allBfObjects,
```
It feels unnatural that `makeTableCorBayes` doesn't return.. a table?

Is there a reason you're creating one massive state object? Because you don't need to do that anymore (you can create as many state objects as you'd like)... It's great because then you can compartmentalize your code and don't couple everything so tightly.

TimKDJ commented on JASP-Engine/JASP/R/correlationbayesian.R:
```diff
@@ -15,2850 +15,1312 @@
+
+  # e. Fill ----
+  #
+  allBfObjects <- .fillCorBayesTable("table"=corBayesTable, "options"=options, 
```
Same here, what does the filling of a table have to do with this object?
Here I'd expect a table to be returned or nothing (because of pass by reference of the table)

TimKDJ reviewed (changes requested):
Hey there it is! :)

Didn't encounter a single crash while testing -- nice. Did note the following though:
- [x] To be consistent can you use long dashes in the matrix table just like it used to be + how frequentist correlation is now (`"\u2014"`)
- [x] For the flag supported correlation you should do a set footnote without a message to prevent additional footnotes from showing up under the table (see https://github.com/jasp-stats/INTERNAL-jasp/issues/562):
![corRewriteFootnote](https://user-images.githubusercontent.com/18737241/69170784-f4992380-0afa-11ea-9bb5-e47049ee4f1f.png)
- [x] "Density" label should be smaller in bayesian correlation matrix. It's disproportionally large in comparison to the legend and almost overlaps:
![corRewriteAxisLabel](https://user-images.githubusercontent.com/18737241/69170764-efd46f80-0afa-11ea-9c82-57ec598dd33b.png)
- [x] I think it might bemore sensible to set the source of the individual pairs to be the variables entered in the main analysis (otherwise people have to search multiple times): `AvailableVariablesList { name: "allVariablesList2" source: "variables" }`
- [x] Messages in prior posterior plot look a lot different than robustness and sequential analysis:
![corRewriteErrMsg](https://user-images.githubusercontent.com/18737241/69171067-8012b480-0afb-11ea-8b7a-ca0a6bd4e838.png)
- [x] In the scatter plot it would be nice if we could get the variables as labels on the axes again (like it used to be), so the plots can be used without needing the var1-var2 context of the JASP container
- [x] Color legend of robustness plot is not correct (white/black in legend and red/gray in plot):
![corRewriteRobustness](https://user-images.githubusercontent.com/18737241/69171288-f0b9d100-0afb-11ea-8893-e04847a865b6.png)
- [ ] The analysis is slower, but not that big of a deal (e.g., switching from missing values type used to be recalculated much quicker). Maybe worthwhile to profile it at some point.

vandenman committed:
Rewrite NetworkAnalysis (#3657)

* Rewrite NetwoekAnalysis

* fixed comments

* removed derivedOptions

* incremented position of plotContainer

TimKDJ committed:
Toggle the status indicator as soon as an analysis status changes (#3666)

JorisGoosen committed:
Switched back to Qt 5.12.6

Rewrote ColumnsModel to not need QTransposeProxyModel
Also fixed a small bug where changin a columntype to scale went unnoticed by qml...

JorisGoosen committed:
Get version for update from static.jasp-stats.org

Implements https://github.com/jasp-stats/INTERNAL-jasp/issues/589

Commit:
All help files are missing (#3667)

Solves https://github.com/jasp-stats/INTERNAL-jasp/issues/585

vandenman committed:
JASPgraphs now passes R CMD check (#3661)

* started passing elements of R CMD check and deprecated unused functions

* fixed all errors and notes except the warning about undocumented objects

* Now passing R CMD check

* export subset methods on JASPgraphsplot

* increment version number

* fix failing test

* check if bfpizza labels need parsing

JorisGoosen committed:
Separators were a bit broken in dynamic module

JorisGoosen committed:
https://www.youtube.com/watch?v=M-b3iU-INDo

JorisGoosen committed:
Engine now recovers "gracefully"

Implements https://github.com/jasp-stats/INTERNAL-jasp/issues/591

JorisGoosen committed:
Upping the jasp-r-interface version

vandenman committed:
FormulaField for math expressions (#3670)

* tempcommit

* FormulaField now functional

* tweaked R api and implemented FormulaField in binomialttestbayesian.R

* fixed unit tests

JorisGoosen committed:
Improvements to css for results

DarkTheme now has better images and plots are inverted
Both themes now have opacity lowered for non-selected analyses so that it is clear what is selected.

JorisGoosen committed:
Saving plot as png fixed

JorisGoosen committed:
Encoding column names changed

- Change encoding of columnNames to something less complicated.
- Every column is now simply encoded to JaspColumn_.#._Encoded and no nonsense with base64 and all that.
- R uses the same code as C++. (issue: https://github.com/jasp-stats/INTERNAL-jasp/issues/543)
- This Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/353 (en/decoding cyrillic doesn't work)
-.v and .unv have been turned off though because of moving encoding/decoding out of R Issue https://github.com/jasp-stats/INTERNAL-jasp/issues/95
- new encoding and decoding functions have been added under columnEncoding.R and should also work in jaspTools (by not encoding and decoding then)
- Plots however now are messed up... Because they do not decode anything the changed names are shown, this should be fixed in JaspGraphs and manually in basePlots?
- Should also fix issue with ordering of dataset columns (https://github.com/jasp-stats/jasp-issues/issues/41)
- A sort of meta entry has been added to the options to tell the engine what it should expect to be columnNames
- supported added for encoding and decoding of plots by @vandenman
-- some analyses still need manual decoding
-- png hacks ftw
- Made automatic encoding of everything optional with a define in JASP.pri

Also made a small change to the R side of .readDataSet... To remove NA's and make it slightly less stupid

JorisGoosen committed:
Some tweaks for darkTheme

Also added small message to define for automatic encoding of columnnames

AlexanderLyNL committed:
Merge branch 'corrieresults' of https://github.com/AlexanderLyNL/jasp-desktop into corrieresults

AlexanderLyNL committed:
Fixed up most of Tim's requests

- TODO(Alexander) package
